backend/pystac_io_methods.py

```diff
@@ -16,14 +17,16 @@
 S3_CLIENT: S3Client = boto3.client("s3", config=CONFIG)
 
 
-class S3StacIO(StacIO):  # type: ignore[misc] # https://github.com/stac-utils/pystac/issues/582
+class S3StacIO(StacIO):
     def read_text(self, source: Union[str, Link], *_args: Any, **_kwargs: Any) -> str:
-        bucket, key = get_bucket_and_key_from_url(source)
+        url = source.href if isinstance(source, Link) else source
+        bucket, key = get_bucket_and_key_from_url(url)
```

Comment:
get_bucket_and_key_from_url expects a str so using source.href if it is of type Link. Unsure if this is appropriate

Reply:
That's fine; it's basically what they do in `StacIO.read_stac_object`.